guest/rust/api_core/src/client/mesh.rs

```diff
@@ -0,0 +1,42 @@
+use crate::global::{ProceduralMeshHandle, Vec2, Vec3};
+use crate::internal::conversion::*;
+use crate::internal::wit;
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct Vertex {
+    pub position: Vec3,
+    pub normal: Vec3,
+    pub tangent: Vec3,
+    pub texcoord0: Vec2,
+}
+
+#[derive(Clone)]
+pub struct Descriptor<'a> {
+    pub vertices: &'a [Vertex],
+    pub indices: &'a [u32],
+}
+
+impl<'a> IntoBindgen for Descriptor<'a> {
+    type Item = wit::client_mesh::Descriptor<'a>;
+
+    fn into_bindgen(self) -> Self::Item {
+        Self::Item {
+            vertices: unsafe {
+                std::slice::from_raw_parts(
+                    self.vertices.as_ptr().cast::<wit::client_mesh::Vertex>(),
```

Comment:
For size I can add a static assert. For type layout I think we can rely on `#[repr(C)]` doing the heavy lifting. Since all participating types are `#[repr(C)]` I think it's ok.

Reply:
The best solution would be making `WIT` types implement `bytemuck::Pod` and `bytemuck::Zeroed` if they can.